OutOfSchool/OutOfSchool.WebApi.Tests/Services/Database/SensitiveWorkshopsServiceDBTests.cs

```diff
@@ -0,0 +1,211 @@
+        mapper = TestHelper.CreateMapperInstanceOfProfileTypes<CommonProfile, MappingProfile>();
+        providerAdminRepository = new Mock<IProviderAdminRepository>();
+        averageRatingServiceMock = new Mock<IAverageRatingService>();
+        providerRepositoryMock = new Mock<IProviderRepository>();
+        currentUserServiceMock = new Mock<ICurrentUserService>();
+        ministryAdminServiceMock = new Mock<IMinistryAdminService>();
+        regionAdminServiceMock = new Mock<IRegionAdminService>();
+        codeficatorServiceMock = new Mock<ICodeficatorService>();
+
+        sensitiveWorkshopService =
+            new WorkshopService(
+                workshopRepository,
+                dateTimeRangeRepository.Object,
+                roomRepository.Object,
+                teacherService.Object,
+                logger.Object,
+                mapper,
+                workshopImagesMediator.Object,
+                providerAdminRepository.Object,
+                averageRatingServiceMock.Object,
+                providerRepositoryMock.Object,
+                currentUserServiceMock.Object,
+                ministryAdminServiceMock.Object,
+                regionAdminServiceMock.Object,
+                codeficatorServiceMock.Object);
+
+        institutionId = Guid.Parse("f72f6404-0d4a-4a6a-bee1-f1a660f0f556");
+        Seed();
+    }
+
+    [TearDown]
+    public void Dispose()
+    {
+        dbContext.Dispose();
+    }
+
+    [Test]
+    public async Task FetchByFilterForAdmins_FilterIsNull_ShouldBuildPredicateAndReturnAllEntities()
+    {
+        // Arrange
+        var workshopDto = await MapWorkshopsToDtos();
+        var expectedResult = new SearchResult<WorkshopDto>()
+        {
+            TotalAmount = workshopDto.Count,
+            Entities = workshopDto,
+        };
+
+        // Act
+        var result = await sensitiveWorkshopService.FetchByFilterForAdmins().ConfigureAwait(false);
+
+        // Assert
+        result.Should().BeEquivalentTo(expectedResult);
+    }
+
+    [Test]
+    public async Task FetchByFilterForAdmins_ByFilterCriteria_ShouldBuildPredicateAndReturnMatchEntities()
+    {
+        // Arrange
+        var workshopDto = await MapWorkshopsToDtos();
+        var filter = new WorkshopFilterAdministration()
+        {
+            InstitutionId = institutionId,
+            SearchString = "keyWord",
+            Size = 100,
+        };
+        var expectedEntities = new List<WorkshopDto> {workshopDto[0] };
+
+        var expectedResult = new SearchResult<WorkshopDto>()
+        {
+            TotalAmount = expectedEntities.Count,
+            Entities = expectedEntities,
+        };
+
+        // Act
+        var result = await sensitiveWorkshopService.FetchByFilterForAdmins(filter).ConfigureAwait(false);
+
+        // Assert
+        result.Should().BeEquivalentTo(expectedResult);
+    }
+
+    private void Seed()
```

Comment:
(Tests set) Could you add few tests for different "CATOTGId" cases? Different "InstitutionId" cases?  Combinations of them?

At the moment I see only limited coverage of filtering logic.

Reply:
Thank you, I will correct it